src/main/java/org/generationcp/middleware/dao/GermplasmDAO.java

```diff
@@ -497,6 +497,31 @@ public Germplasm getByGIDWithPrefAbbrev(final Integer gid) {
 		return new ArrayList<>();
 	}
 
+	/**
+	 * Only returns germplasm with progeny (Germplasm Id is assigned to other germplasm's gpid1, gpid2 OR progntrs.pid)
+	 *
+	 * @param gids - gids to be checked for descendants
+	 * @return
+	 */
+	public List<Germplasm> getGermplasmWithDescendants(final Set<Integer> gids) {
```

Comment:
As commented in service seems we can only return GIDs instead of the whole entity